From 875b2414cd68df64aeead651a9b05ae9bc3d88ef Mon Sep 17 00:00:00 2001
From: Florian Weimer <fweimer@redhat.com>
Date: Thu, 4 Aug 2022 17:54:48 +0200
Subject: [PATCH 303/423] dlfcn: Pass caller pointer to static dlopen
 implementation (bug 29446)

Fixes commit 0c1c3a771eceec46e66ce1183cf988e2303bd373 ("dlfcn: Move
dlopen into libc").

(cherry picked from commit ed0185e4129130cbe081c221efb758fb400623ce)
---
 NEWS           | 1 +
 dlfcn/dlopen.c | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/NEWS b/NEWS
index 04fe72b5f6..71a3ab0ed4 100644
--- a/NEWS
+++ b/NEWS
@@ -112,6 +112,7 @@ The following bugs are resolved with this release:
   [29211] libc: __open_catalog is not y2038 aware
   [29213] libc: gconv_parseconfdir is not y2038 aware
   [29214] nptl: pthread_setcanceltype fails to set type
+  [29446] _dlopen now ignores dl_caller argument in static mode
 
 
 Version 2.34
diff --git a/dlfcn/dlopen.c b/dlfcn/dlopen.c
index 9c59c751c4..739d17baaf 100644
--- a/dlfcn/dlopen.c
+++ b/dlfcn/dlopen.c
@@ -90,7 +90,7 @@ compat_symbol (libdl, ___dlopen, dlopen, GLIBC_2_1);
 void *
 __dlopen (const char *file, int mode, void *dl_caller)
 {
-  return dlopen_implementation (file, mode, RETURN_ADDRESS (0));
+  return dlopen_implementation (file, mode, dl_caller);
 }
 
 void *
-- 
2.30.2

